integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDynamicDomainInImage.java

```diff
@@ -0,0 +1,309 @@
+   * @param replicaCount count of replicas
+   * @param miiImage name of model in image
+   */
+  private void createDomainCrAndVerify(
+      String domainUid, String domNamespace, String adminSecretName,
+      String repoSecretName, String encryptionSecretName, String rcuAccessSecretName,
+      String opssWalletPasswordSecretName, int replicaCount, String miiImage) {
+    // create the domain CR
+    Domain domain = new Domain()
+            .apiVersion(DOMAIN_API_VERSION)
+            .kind("Domain")
+            .metadata(new V1ObjectMeta()
+                .name(domainUid)
+                .namespace(domNamespace))
+            .spec(new DomainSpec()
+                .domainUid(domainUid)
+                .domainHomeSourceType("FromModel")
+                .image(miiImage)
+                .imagePullPolicy("IfNotPresent")
+                .addImagePullSecretsItem(new V1LocalObjectReference()
+                    .name(repoSecretName))
+                .webLogicCredentialsSecret(new V1SecretReference()
+                    .name(adminSecretName)
+                    .namespace(domNamespace))
+                .includeServerOutInPodLog(true)
+                .serverStartPolicy("IF_NEEDED")
+                .serverPod(new ServerPod()
+                    .addEnvItem(new V1EnvVar()
+                        .name("ALLOW_DYNAMIC_CLUSTER_IN_FMW")
+                        .value("true"))
+                    .addEnvItem(new V1EnvVar()
+                        .name("JAVA_OPTIONS")
+                        .value("-Dweblogic.StdoutDebugEnabled=false"))
+                    .addEnvItem(new V1EnvVar()
+                        .name("USER_MEM_ARGS")
+                        .value("-Djava.security.egd=file:/dev/./urandom ")))
+                .adminServer(new AdminServer()
+                    .serverStartState("RUNNING")
+                    .adminService(new AdminService()
+                        .addChannelsItem(new Channel()
+                            .channelName("default")
+                            .nodePort(0))))
+                .addClustersItem(new Cluster()
+                    .clusterName("cluster-1")
+                    .replicas(replicaCount)
+                    .serverStartState("RUNNING"))
+                .configuration(new Configuration()
+                    .opss(new Opss()
+                        .walletPasswordSecret(opssWalletPasswordSecretName))
+                    .model(new Model()
+                        .domainType("JRF")
+                        .runtimeEncryptionSecret(encryptionSecretName))
+                    .addSecretsItem(rcuAccessSecretName)
+                    .introspectorJobActiveDeadlineSeconds(600L)));
+
+    createDomainAndVerify(domain, domainNamespace);
+  }
+
+  /**
+   * Verify Pod is ready and service exists for both admin server and managed servers.
+   * Verify EM console is accessible.
+   */
+  private void verifyDomainReady() {
```

Comment:
Can you embed the content of the methods verifyDomainReady() to test Method for clarity.

Reply:
I prefer a helper method to have the test method clean and simple, if you are ok about it

Reply:
Helper method is needed when we use repetitive code. Here I see one instance of the method  verifyDomainReady() in the class